drivers/i2c/i2c_esp32.c

```diff
@@ -591,9 +591,8 @@ static int IRAM_ATTR i2c_esp32_transfer(const struct device *dev, struct i2c_msg
 	return ret;
 }
 
-static void IRAM_ATTR i2c_esp32_isr(void *arg)
+static void IRAM_ATTR i2c_esp32_isr(const struct device *dev)
```

Comment:
Both changes in ESP32 (counter and I2C) will add build warnings with this change. `esp_intr_alloc` requires `void *arg` in its signature. Can this be removed from this PR?

Reply:
sure